frontend/public/components/monitoring/alert-manager-config.tsx

```diff
@@ -21,32 +27,196 @@ const AlertRouting: React.FC<AlertRoutingProps> = ({ config, secret }) => {
       <div className="row">
         <div className="col-sm-6">
           <dt>Group By</dt>
-          <dd>{_.join(_.get(config, ['route', 'group_by']), ', ')}</dd>
+          <dd>{_.isEmpty(groupBy) ? '-' : _.join(groupBy, ', ')}</dd>
           <dt>Group Wait</dt>
-          <dd>{_.get(config, ['route', 'group_wait'])}</dd>
+          <dd>{_.get(config, ['route', 'group_wait'], '-')}</dd>
         </div>
         <div className="col-sm-6">
           <dl className="co-m-pane__details">
             <dt>Group Interval</dt>
-            <dd>{_.get(config, ['route', 'group_interval'])}</dd>
+            <dd>{_.get(config, ['route', 'group_interval'], '-')}</dd>
             <dt>Repeat Interval</dt>
-            <dd>{_.get(config, ['route', 'repeat_interval'])}</dd>
+            <dd>{_.get(config, ['route', 'repeat_interval'], '-')}</dd>
           </dl>
         </div>
       </div>
-    </React.Fragment>
+    </div>
+  );
+};
+
+const tableColumnClasses = [
+  classNames('col-lg-3', 'col-md-3', 'col-sm-6', 'col-xs-6'),
+  classNames('col-lg-3', 'col-md-3', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-6', 'col-md-6', 'col-sm-6', 'col-xs-6'),
+];
+
+const ReceiverTableHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Integration Type',
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Routing Labels',
+      props: { className: tableColumnClasses[2] },
+    },
+  ];
+};
+ReceiverTableHeader.displayName = 'ReceiverTableHeader';
+
+const getIntegrationTypes = (receiver) => {
+  /* Given receiver = {
+       "name": "team-X-pager",
+       "email_configs": [...],
+       "pagerduty_configs": [...]
+     };
+     returns "email, pagerduty"
+  */
+  const integrationTypes = _.filter(_.keys(receiver), (key) => _.includes(key, '_configs'));
+  return _.join(_.map(integrationTypes, (type) => type.substr(0, type.indexOf('_configs'))), ', ');
+};
+
+// Recursive function to get hierarchy of routing labels for each receiver
+const getRoutinglabels = (routes, parentLabels) => {
```

Comment:
Nit: Should be `getRoutingLabels` (upper case `L`)

Reply:
fixed